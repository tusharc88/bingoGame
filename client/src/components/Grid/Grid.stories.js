import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { withKnobs, number } from '@storybook/addon-knobs';
import Grid from './index';

storiesOf('Grid', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo('default', <Grid matrixNum={number('matrixNum', 5)} />, () => (
    <Grid matrixNum={number('matrixNum', 5)} />
  ));
