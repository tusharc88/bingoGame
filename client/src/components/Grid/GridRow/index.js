import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Cell from '../../Cell/index';

const GridRowWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

const GridRow = ({ matrixNum, rowNum }) => {
  const cols = [];
  for (let i = 0; i < matrixNum; i += 1) {
    cols.push(
      <Cell currentNumber={1} key={`${rowNum}-${i}`}>
        {1}
      </Cell>
    );
  }

  return <GridRowWrapper>{cols}</GridRowWrapper>;
};

GridRow.propTypes = {
  matrixNum: PropTypes.number.isRequired,
  rowNum: PropTypes.number.isRequired,
};

export default GridRow;
