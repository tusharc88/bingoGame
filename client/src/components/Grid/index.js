import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Cell from '../Cell/index';

const GridWrapper = styled.div`
  transition: all 0.3s ease-in-out;
  transition-property: transform box-shadow;
  &:hover {
    box-shadow: 0px 5px 5px 0px #aeb0b1;
    transform: translateY(-3px);
    transition: all 0.3s ease-in-out;
    transition-property: transform box-shadow;
  }
`;

const GridRowWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

// TODO: Get currentNumber from express call,
// pass it to Cell using React Context Api
// Get child number for each cell using random number generator, save in redux
// check for number should not be repeated efficiently
export const Grid = ({ matrixNum }) => (
  <GridWrapper>
    {Array.from({ length: matrixNum }, (rowVal, rowKey) => (
      <GridRowWrapper key={rowKey}>
        {Array.from({ length: matrixNum }, (colVal, colKey) => (
          <Cell currentNumber={1} key={`${rowKey}-${colKey}`}>
            {1}
          </Cell>
        ))}
      </GridRowWrapper>
    ))}
  </GridWrapper>
);

Grid.propTypes = {
  matrixNum: PropTypes.number.isRequired,
};

export default Grid;
