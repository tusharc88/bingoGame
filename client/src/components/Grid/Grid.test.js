import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import Grid from './index';

const wrap = (props = {}) => shallow(<Grid matrixNum={5} {...props} />);

test('Renders with default props', () => {
  const wrapper = wrap();
  const tree = toJSON(wrapper);
  expect(tree).toMatchSnapshot();
});
