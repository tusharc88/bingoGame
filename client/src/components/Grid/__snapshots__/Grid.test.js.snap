// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
<styled.div>
  <styled.div
    key="0"
  >
    <Cell
      currentNumber={1}
      key="0-0"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="0-1"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="0-2"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="0-3"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="0-4"
    >
      1
    </Cell>
  </styled.div>
  <styled.div
    key="1"
  >
    <Cell
      currentNumber={1}
      key="1-0"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="1-1"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="1-2"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="1-3"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="1-4"
    >
      1
    </Cell>
  </styled.div>
  <styled.div
    key="2"
  >
    <Cell
      currentNumber={1}
      key="2-0"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="2-1"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="2-2"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="2-3"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="2-4"
    >
      1
    </Cell>
  </styled.div>
  <styled.div
    key="3"
  >
    <Cell
      currentNumber={1}
      key="3-0"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="3-1"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="3-2"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="3-3"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="3-4"
    >
      1
    </Cell>
  </styled.div>
  <styled.div
    key="4"
  >
    <Cell
      currentNumber={1}
      key="4-0"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="4-1"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="4-2"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="4-3"
    >
      1
    </Cell>
    <Cell
      currentNumber={1}
      key="4-4"
    >
      1
    </Cell>
  </styled.div>
</styled.div>
`;
